app/services/ial_context.rb

```diff
@@ -18,11 +19,20 @@ def default_to_ial2?
     ial.nil? && ial2_service_provider?
   end
 
+  def user_ial2_verified?
+    user&.active_profile&.verified_at != nil
+  end
+
   def ialmax_requested?
     ial&.zero?
   end
 
+  def bill_for_ial_1_or_2
+    ial2_or_greater? ? 2 : 1
+  end
+
   def ial2_requested?
+    return true if ialmax_requested? && user_ial2_verified?
     ial == ::Idp::Constants::IAL2
```

Comment:
no need for an early return IMO

```suggestion
    (ialmax_requested? && user_ial2_verified?) || ial == ::Idp::Constants::IAL2
```